src/librustc_typeck/mem_categorization.rs

```diff
@@ -515,9 +525,19 @@ impl<'a, 'tcx> MemCategorizationContext<'a, 'tcx> {
         base_place: PlaceWithHirId<'tcx>,
         ty: Ty<'tcx>,
     ) -> PlaceWithHirId<'tcx> {
+        let base_ty = base_place.place.ty();
```

Comment:
rename this to `base_final_ty`, or something similar? It's a little confusing becuase place also has the `base_ty`, which is essentially the opposite.

Reply:
Done.